### PR TITLE
Attach manifold

### PR DESCRIPTION
**Pull request by ferrine**
Some manifold methods output values lying on the manifold. This PR makes them of type ManifoldTensor

### CODE REVIEW

File: geoopt/tensor.py [hunk 1]
```diff
@@ -24,6 +24,7 @@ def __new__(
         if len(args) == 1 and isinstance(args[0], torch.Tensor):
             data = args[0].data
         else:
+            print(args)
```
Comment (rrkarim) [inline geoopt/tensor.py]: remove print

**Comment by rrkarim** [comment]
@ferrine I'm looking at this PR, need some time to understand geoopt-layers too.

**Comment by ferrine** [comment]
Sure, I'm busy with other stuff currently, go ahead

пт, 8 нояб. 2019 г., 22:19 Rasul Karimov <notifications@github.com>:

File: geoopt/manifolds/scaled.py [hunk 2]
```diff
@@ -26,23 +31,31 @@ def stub(self, *args, **kwargs):
...
-                (
-                    rescale_value(res, self.scale, power)
-                    for res, power in zip(results, scaling_info.results)
-                )
+            arguments[k] = rescale_value(arguments[k], self.scale, power, None, False)
+        params = params.__class__(signature, arguments)
+        results = function(*params.args, **params.kwargs)
+        if not scaling_info.results and not scaling_info.attach:
+            # do nothing, attach if needed
+            return results if 0 not in scaling_info.attach else self.attach(results)
```
Comment (rrkarim) [inline geoopt/manifolds/scaled.py]: This line is pretty obscured for me.

**Comment by ferrine** [comment]
This PR is not yet ready to ship. There are unresolved issues with shapes that should be resolved first.

**Pull request closed without merging**